tool/test/license_utils_test.dart

```diff
@@ -0,0 +1,667 @@
+      final config = LicenseConfig.fromYamlFile(configFile);
+      final header = LicenseHeader();
+
+      final contentsBeforeUpdate = testFile1.readAsStringSync();
+      final results = await header.bulkUpdate(
+        directory: testDirectory,
+        config: config,
+      );
+      final contentsAfterUpdate = testFile1.readAsStringSync();
+
+      final includedPaths = results.includedPaths;
+      expect(includedPaths, isNotNull);
+      expect(includedPaths.length, equals(7));
+      // Order is not guaranteed
+      expect(includedPaths.contains(testFile1.path), true);
+      expect(contentsBeforeUpdate, isNot(equals(contentsAfterUpdate)));
+      expect(includedPaths.contains(testFile2.path), true);
+      expect(includedPaths.contains(testFile3.path), true);
+      expect(includedPaths.contains(testFile7.path), true);
+      expect(includedPaths.contains(testFile8.path), true);
+      expect(includedPaths.contains(testFile9.path), true);
+      expect(includedPaths.contains(testFile10.path), true);
+
+      final updatedPaths = results.updatedPaths;
+      expect(updatedPaths, isNotNull);
+      // testFile9 and testFile10 are intentionally misconfigured and so they
+      // won't be updated even though they are on the include list.
+      expect(updatedPaths.length, equals(5));
+      // Order is not guaranteed
+      expect(updatedPaths.contains(testFile1.path), true);
+      expect(updatedPaths.contains(testFile2.path), true);
+      expect(updatedPaths.contains(testFile3.path), true);
+      expect(updatedPaths.contains(testFile7.path), true);
+      expect(updatedPaths.contains(testFile8.path), true);
+    });
+
+    test('license headers bulk update can be dry run', () async {
+      await _setupTestConfigFile();
+      final config = LicenseConfig.fromYamlFile(configFile);
+      final header = LicenseHeader();
+
+      final contentsBeforeUpdate = testFile1.readAsStringSync();
+      final results = await header.bulkUpdate(
+        directory: testDirectory,
+        config: config,
+        dryRun: true,
+      );
+      final contentsAfterUpdate = testFile1.readAsStringSync();
+
+      final updatedPaths = results.updatedPaths;
+      expect(updatedPaths, isNotNull);
+      expect(updatedPaths.length, equals(5));
+      expect(updatedPaths.contains(testFile1.path), true);
+      expect(contentsBeforeUpdate, equals(contentsAfterUpdate));
+    });
+  });
+
+  test(
+    'repo wide check',
+    () async {
+      // This test is currently skipped because not all existing files
+      // have had their license headers updated to the correct license text.
+      // So this test will always fail. Set skip to false to run locally, but
+      // don't commit the change to the repository.
+
+      // TODO(mossmana): This test should stop being skipped only when it is safe to check just new files going forward.
+      final rootPathMatcher = RegExp(r'(.*[/|\\]devtools[/|\\]).*');
+      // TODO(mossmana): make this work on Google3
```

Comment:
we actually don't run DevTools tests in google3 other than a smoke test to ensure it builds